src/main.js

```diff
@@ -93,7 +94,16 @@ module.exports = {
       }
     }))
 
-    this.subscriptions.add(atom.commands.add('atom-text-editor', {
+    // Marks each JavaScript editor with a CSS class so that
+    // we can enable commands only for JavaScript editors.
+    this.subscriptions.add(atom.workspace.observeTextEditors((textEditor) => {
+      if (textEditor.getRootScopeDescriptor().scopes
```

Comment:
That's actually the _upside_, not downside. Say for example you are editing an HTML file with some embedded JS code. The scope of the file would be `source.html`, while the scope of the embedded JS will be `source.js.embedded.html`. If we just went with your current method the file would never trigger linting on the embedded JS code, while checking the cursor means that if the user has a cursor in a scope that the linter provider understands it will be enabled.

This is how `linter` itself triggers providers in the first place, and one of the reasons I'm not exactly sold on the whole idea of adding a class to the editor instead of just checking things when called since we aren't really sure what situations the user can get into.

For other linters where there is a nice pre-defined list of possible scopes it might work, but for here I'm not sure I see how it will work without a ton of hacky workarounds (observing all editor cursors and checking if one ever enters a supported scope in the editor...).